lib/season_stats_module.rb

```diff
@@ -0,0 +1,170 @@
+module SeasonStatistics
+  require 'pry'
+
+  def find_all_games_from_season(season_id)
+    @games.find_all do |game|
+      game.season == season_id
+    end
+  end
+
+  def game_teams_data_for_season(season_id)
+    @game_teams.find_all do |game|
+      game.game_id[0..3] == season_id[0..3]
+    end
+  end
+
+  def season_coaches(season_id)
+    game_teams_data_for_season(season_id).map do |game|
+      game.head_coach
+    end.uniq
+  end
+
+  def winningest_coach(season_id)
+    coaches_hash = Hash.new
+    season_coaches(season_id).find_all do |coach|
+
+      total_games = game_teams_data_for_season(season_id).count do |game|
+        game.head_coach == coach
+      end
```

Comment:
I like how many helper methods you are using. It will helps make code more concise